tests/unittests/test_datasource/test_common.py

```diff
@@ -95,5 +97,14 @@ def test_data_sources_have_valid_network_config_sources(self):
                 self.assertTrue(hasattr(sources.NetworkConfigSource, cfg_src),
                                 fail_msg)
 
+    def test_expected_dsname_defined(self):
+        for ds in DEFAULT_LOCAL + DEFAULT_NETWORK:
+            fail_msg = (
+                '{} has an invalid / missing dsname property: {}'.format(
+                    str(ds), str(ds.dsname)
+                )
+            )
+            self.assertNotEqual(ds.dsname, DataSource.dsname, fail_msg)
+            self.assertIsNotNone(ds.dsname, None)
```

Comment:
This still isn't quite right:

```suggestion
            self.assertIsNotNone(ds.dsname)
```